lib/utils/search.js

```diff
@@ -80,4 +80,38 @@ const isJSON = text => {
   return true;
 };
 
-export { getSearchResults };
+const capitalise = phrase => {
+  return phrase
+    .split(' ')
+    .map(word => word.charAt(0).toUpperCase() + word.slice(1))
+    .join(' ');
+};
+
+const getDifferentCasing = array => {
+  return array
+    .concat(array.map(word => capitalise(word)))
+    .concat(array.map(word => word.toUpperCase()));
+};
+
+const getWordsToHighlight = searchTerm => {
+  const synonyms = isJSON(process.env.SYNONYMS) ? JSON.parse(process.env.SYNONYMS) : '{}';
+
+  let normalisedSearchTerm = normalise(searchTerm.trim());
+  let words = removeCommonWords(normalisedSearchTerm.split(' '));
+
+  let highlightables = getDifferentCasing([normalisedSearchTerm]);
+
+  highlightables[0] !== words[0]
+    ? (highlightables = highlightables.concat(getDifferentCasing(words)))
+    : null;
+
+  const foundSynonyms = highlightables
+    .map(highlightable => Object.keys(synonyms).find(key => synonyms[key].includes(highlightable)))
+    .filter(x => x);
+
+  highlightables = highlightables.concat(getDifferentCasing(foundSynonyms));
+
+  return highlightables.filter(x => x != '');
+};
+
+export { getSearchResults, getWordsToHighlight };
```

Comment:
This is more of a question about the acceptance criteria as I don't think we have them well defined.
If we have a search word `phone` and the description word `telephone`, should there be any highlighting happening?

So we have 3 possible cases:
1) telephone _(no highlight)_
2) tele**phone** _(partial highlight)_
3) **telephone** _(full highlight)_

Looking at how the search works, it seems that it assumes a match under the scenario 3:
`words.filter(word => resourceIndexValues.some(x => x.includes(word)))`, where it's effectively saying:
`find search words that are a substring of service description (& etc.) words`.

I might not be understanding the highlighting code, but it seems as if 3rd scenario or even 2nd scenario is unaccounted for.

Reply:
I tried solving this with that clunky looking dictionary I commited last week [[commit](https://github.com/LBHackney-IT/coronavirus-frontdoor-snapshot/pull/138/commits/40938d3107388be6151d6f73ee23b76778a9fd51#diff-1e838060cbc93993c33d3d7174b025d94772c15737dd28ac4bfae83a5bd1e5f7R71-R74)].

I don't think I went all the way with it though. I still think the solution would be doing the filtering of that dictionary like I did it there, except for that the filtering the way it was done there wasn't correct. What I did was I was filtering by full word matches _(looking for normalised description word within normalised search term words array)_. What I should have done was to filter by any matching sub-string normalised search term word within normalised description words array.

Reply:
Spoke this one through with @kosiakkatrina on Tandem. We identified that this could be a problem under the case, where upon finding the Service Record by matching user search `phone` on service text `telephone`, the highlighting scenario 1 will be executing making the Service show up on search results without having a single piece of text highlighted.

Whether or not it's something that's going to be tackled on this PR is still unclear.